include/xsimd/config/xsimd_arch.hpp

```diff
@@ -89,6 +129,11 @@ namespace xsimd
         {
             (void)std::initializer_list<bool> { (f(Archs {}), true)... };
         }
+
+        static constexpr std::size_t alignment()
+        {
+            return detail::max_alignment<Archs...>::value;
```

Comment:
Side note: theoretically, it should probably be the Least common multiple instead. Because we only have power of twos, max is faster, but maybe worth adding a comment?

Reply:
I think this could ultimately be written as:
```
return details::max_of(Archs::alignment()...);
```

Reply:
alignas() causes a compile error if you give it a value that isn't a power of two (or zero), so I don't think a check is really needed for this (as the rest of the library will fail to compile too).

I've implemented it as you suggested, though I think it's a bit messy to use a constexpr function for this, as you end up having to have a separate type variable (in a parameter pack) for each element, and you can't use temporary variables.

I've put a version using template structs instead here:

https://github.com/xtensor-stack/xsimd/compare/master...tomjnixon:arch_list_alignment_struct

What do you think?